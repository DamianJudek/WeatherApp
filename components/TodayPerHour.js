import React from 'react';
import { StyleSheet } from 'react-native';
import PropTypes from 'prop-types';
import Hour from './Hour';
import FadeInView from './FadeInView';

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    marginBottom: 10,
    borderRadius: 50,

    backgroundColor: 'rgba(255, 255, 255, 0.3)',
    // backgroundColor: 'rgba(30, 30, 30, 0.4)',
//tutaj bylo wczesniej
//     backgroundColor: 'rgba(220, 220, 220, 0.2)',


    padding: 15,
  },
  section: {
    flex: 1,
    borderRightWidth: 1,
    borderColor: '#fff',
    justifyContent: 'center',
    alignItems: 'center',
  },
  lastSection: {
    borderRightWidth: 0,
  },
});


const TodayPerHour = ({ weather }) => {
  
  return (
    <FadeInView delay={300} style={styles.container}>
      <Hour weather={weather[0]} style={styles.section} />
      <Hour weather={weather[1]} style={styles.section} />
      <Hour weather={weather[2]} style={styles.section} />
      <Hour weather={weather[3]} style={[styles.section, styles.lastSection]} />
    </FadeInView>
  );
};

TodayPerHour.propTypes = {
  weather: PropTypes.array,
};

export default TodayPerHour;
