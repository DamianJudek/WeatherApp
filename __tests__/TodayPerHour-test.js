import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import TodayPerHour from '../components/TodayPerHour';

jest.mock('react-native/Libraries/Animated/src/NativeAnimatedHelper');
test('renders correctly', () => {
  const snap = renderer
    .create(
      <TodayPerHour
        weather={[
          {
            dayNumber: 5,
            description: 'clouds',
            hour: '21:00',
            humidity: 87,
            icon: '03n',
            temp: 17,
            wind: 2.22,
          },
          {
            dayNumber: 6,
            description: 'clouds',
            hour: '00:00',
            humidity: 91,
            icon: '04n',
            temp: 16,
            wind: 2.78,
          },
          {
            dayNumber: 6,
            description: 'rain',
            hour: '03:00',
            humidity: 93,
            icon: '03n',
            temp: 15,
            wind: 3.02,
          },
          {
            dayNumber: 6,
            description: 'rain',
            hour: '06:00',
            humidity: 88,
            icon: '03d',
            temp: 17,
            wind: 2.93,
          },
        ]}
      />,
    )
    .toJSON();
  expect(snap).toMatchSnapshot();
});
