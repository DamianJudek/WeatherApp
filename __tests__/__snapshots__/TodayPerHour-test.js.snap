// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "rgba(255, 255, 255, 0.3)",
      "borderRadius": 50,
      "flex": 1,
      "flexDirection": "row",
      "marginBottom": 10,
      "opacity": 0,
      "padding": 15,
      "transform": Array [
        Object {
          "scale": 0.8,
        },
      ],
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderRightWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 35,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
        }
      }
    >
      21:00
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "http://openweathermap.org/img/wn/03n@2x.png",
        }
      }
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
        }
      }
    >
      17℃
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderRightWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 35,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
        }
      }
    >
      00:00
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "http://openweathermap.org/img/wn/04n@2x.png",
        }
      }
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
        }
      }
    >
      16℃
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderRightWidth": 1,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 35,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
        }
      }
    >
      03:00
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "http://openweathermap.org/img/wn/03n@2x.png",
        }
      }
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
        }
      }
    >
      15℃
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "borderColor": "#fff",
        "borderRightWidth": 0,
        "flex": 1,
        "justifyContent": "center",
        "opacity": 0,
        "transform": Array [
          Object {
            "translateY": 35,
          },
        ],
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
        }
      }
    >
      06:00
    </Text>
    <Image
      resizeMode="contain"
      source={
        Object {
          "uri": "http://openweathermap.org/img/wn/03d@2x.png",
        }
      }
      style={
        Object {
          "height": 60,
          "width": 60,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#fff",
          "fontFamily": "Roboto",
          "fontSize": 18,
        }
      }
    >
      17℃
    </Text>
  </View>
</View>
`;
